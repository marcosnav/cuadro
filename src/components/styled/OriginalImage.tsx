import styled from 'styled-components';

interface IProps {
  image: string;
  show?: boolean;
}

const OriginalImage = styled.div<IProps>`
  background-image: url("${({ image }) => image}");
  background-size: cover cover;
  background-position: center center;
  height: 100%;
  opacity: ${({ show }) => show ? '1' : '0'};
  pointer-events: ${({ show }) => show ? 'auto' : 'none'};
  position: relative;
  transition: opacity 0.6s ease;
  width: 100%;

  &::before {
    background: ${({ theme }) => theme.SOFT_GRAY};
    content: '';
    box-shadow: inset 0 1px 4px 2px rgba(0, 0, 0, 0.2);
    height: 67px;
    left: 0;
    position: absolute;
    top: 0;
    width: 67px;
  }
`;

OriginalImage.displayName = 'OriginalImage';

export default OriginalImage;
