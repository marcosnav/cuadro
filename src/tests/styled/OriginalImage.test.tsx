import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import OriginalImage from './../../components/styled/OriginalImage';

describe('<OrignalImage />', () => {
  const component = mount(<OriginalImage image={'/cat-image.jpg'} />);

  test('renders properly', () => {
    expect(component).toHaveStyleRule('background-image', 'url("/cat-image.jpg")');
    expect(component).toHaveStyleRule('opacity', '0');
    expect(component).toHaveStyleRule('pointer-events', 'none');
    expect(component).toMatchSnapshot();
  });

  test('show: true', () => {
    component.setProps({ show: true });
    expect(component).toHaveStyleRule('opacity', '1');
    expect(component).toHaveStyleRule('pointer-events', 'auto');
    expect(component).toMatchSnapshot();
  });
});
