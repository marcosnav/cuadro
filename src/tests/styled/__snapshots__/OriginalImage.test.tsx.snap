// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrignalImage /> renders properly 1`] = `
.c0 {
  background-image: url("/cat-image.jpg");
  background-size: cover;
  background-position: center center;
  height: 100%;
  opacity: 0;
  pointer-events: none;
  position: relative;
  -webkit-transition: opacity 0.6s ease;
  transition: opacity 0.6s ease;
  width: 100%;
}

.c0::before {
  background: linear-gradient(30deg,undefined 0%,undefined 100%);
  content: '';
  box-shadow: inset 0 1px 4px 2px rgba(0,0,0,0.2);
  height: 67px;
  left: 0;
  position: absolute;
  top: 0;
  width: 67px;
}

<OriginalImage
  image="/cat-image.jpg"
>
  <div
    className="c0"
  />
</OriginalImage>
`;

exports[`<OrignalImage /> show: true 1`] = `
.c0 {
  background-image: url("/cat-image.jpg");
  background-size: cover;
  background-position: center center;
  height: 100%;
  opacity: 1;
  pointer-events: auto;
  position: relative;
  -webkit-transition: opacity 0.6s ease;
  transition: opacity 0.6s ease;
  width: 100%;
}

.c0::before {
  background: linear-gradient(30deg,undefined 0%,undefined 100%);
  content: '';
  box-shadow: inset 0 1px 4px 2px rgba(0,0,0,0.2);
  height: 67px;
  left: 0;
  position: absolute;
  top: 0;
  width: 67px;
}

<OriginalImage
  image="/cat-image.jpg"
  show={true}
>
  <div
    className="c0"
  />
</OriginalImage>
`;
